pandas/tests/indexes/datetimes/test_date_range.py

```diff
@@ -631,3 +632,58 @@ def test_all_custom_freq(self, freq):
         msg = 'invalid custom frequency string: {freq}'
         with tm.assert_raises_regex(ValueError, msg.format(freq=bad_freq)):
             bdate_range(START, END, freq=bad_freq)
+
+
+class TestTimestampEquivDateRange(object):
+    # Older tests in scalar.test_timestamp.TestTimeSeries constructed
+    # their `stamp` objects using `date_range` instead of the `Timestamp`
+    # constructor.  TestTimestampEquivDateRange checks that these are
+    # equivalent in the pertinent cases.
+
+    def test_date_range_timestamp_equiv(self):
+        rng = date_range('20090415', '20090519', tz='US/Eastern')
+        stamp = rng[0]
+
+        ts = Timestamp('20090415', tz='US/Eastern', freq='D')
+        assert ts == stamp
+
+    def test_date_range_timestamp_equiv_dateutil(self):
+        rng = date_range('20090415', '20090519', tz='dateutil/US/Eastern')
+        stamp = rng[0]
+
+        ts = Timestamp('20090415', tz='dateutil/US/Eastern', freq='D')
+        assert ts == stamp
+
+    def test_date_range_timestamp_equiv_explicit_pytz(self):
+        rng = date_range('20090415', '20090519',
+                         tz=pytz.timezone('US/Eastern'))
+        stamp = rng[0]
+
+        ts = Timestamp('20090415', tz=pytz.timezone('US/Eastern'), freq='D')
+        assert ts == stamp
+
+    def test_date_range_timestamp_equiv_explicit_dateutil(self):
```

Comment:
this can be made a decorator FYI

Reply:
add a note to do this

Reply:
How does the decorator usage work?  grepping didn't turn up any examples.